### ISSUE TITLE
only obtain image features?

### ISSUE THREAD

**Issue by CoinCheung** [facebookresearch/segment-anything-2#11]
Is there any method we can extract features of images ? Like in SAM-v1 where we can only use vit to encode image into features and discard the other parts of the model ?

**Comment by heyoeyo** [comment]
Yes the image features are stored similarly to the v1 model. If you run image prediction like they [describe in the readme](https://github.com/facebookresearch/segment-anything-2?tab=readme-ov-file#image-prediction),
then after the `predictor.set_image(<your_image>)` line, you can access the image features from: `predictor._features`

Though the image encoder now produces 3 feature maps. The one most similar to what the old model outputs is inside: `predictor._features["image_embed"]`, and then there are two other higher-resolution maps inside `predictor._features["high_res_feats"]`
